pkg/controller/useraccount/useraccount_controller_test.go

```diff
@@ -776,6 +771,248 @@ func TestUpdateStatus(t *testing.T) {
+		// given
+		userAcc := newDisabledUserAccount(username, userID)
+		r, req, _ := prepareReconcile(t, username, userAcc, preexistingUser, preexistingNsTmplSet)
+
+		// when
+		res, err := r.Reconcile(req)
+		assert.Equal(t, reconcile.Result{}, res)
+		require.NoError(t, err)
+
+		// then
+		userAcc = &toolchainv1alpha1.UserAccount{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: userAcc.Name, Namespace: "toolchain-member"}, userAcc)
+		require.NoError(t, err)
+		assert.Equal(t, 1, len(userAcc.Status.Conditions))
+		assert.Equal(t, "Disabling", userAcc.Status.Conditions[0].Reason)
+
+		// Check that the associated identity has been deleted
+		// since disabled has been set to true
+		identity := &userv1.Identity{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: ToIdentityName(userAcc.Spec.UserID)}, identity)
+		require.Error(t, err)
+		assert.True(t, apierros.IsNotFound(err))
+
+		// Check that the associated user has been deleted
+		// since disabled has been set to true
+		user := &userv1.User{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: userAcc.Name}, user)
+		require.Error(t, err)
+		assert.True(t, apierros.IsNotFound(err))
+
+		// Get NSTemplate
+		tmplTier := &toolchainv1alpha1.NSTemplateSet{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: userAcc.Name}, tmplTier)
+		require.NoError(t, err)
+	})
+
+	t.Run("deleting disabled useraccount", func(t *testing.T) {
+
+		userAcc := newDisabledUserAccountWithFinalizer(username, userID)
+
+		r, req, _ := prepareReconcile(t, username, userAcc, preexistingNsTmplSet)
+
+		res, err := r.Reconcile(req)
+		assert.Equal(t, reconcile.Result{}, res)
+		require.NoError(t, err)
+
+		// Check that the finalizer is present
+		require.True(t, util.HasFinalizer(userAcc, userAccFinalizerName))
+
+		// Set the deletionTimestamp
+		userAcc.DeletionTimestamp = &metav1.Time{time.Now()} //nolint: govet
+		err = r.client.Update(context.TODO(), userAcc)
+		require.NoError(t, err)
+
+		res, err = r.Reconcile(req)
+		assert.Equal(t, reconcile.Result{}, res)
+		require.NoError(t, err)
+
+		res, err = r.Reconcile(req)
+		assert.Equal(t, reconcile.Result{}, res)
+		require.NoError(t, err)
+
+		res, err = r.Reconcile(req)
+		assert.Equal(t, reconcile.Result{}, res)
+		require.NoError(t, err)
+
+		// Check that the associated identity has been deleted
+		// since disabled has been set to true
+		identity := &userv1.Identity{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: ToIdentityName(userAcc.Spec.UserID)}, identity)
+		require.Error(t, err)
+		assert.True(t, apierros.IsNotFound(err))
+
+		// Check that the associated user has been deleted
+		// since disabled has been set to true
+		user := &userv1.User{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: userAcc.Name}, user)
+		require.Error(t, err)
+		assert.True(t, apierros.IsNotFound(err))
+
+		userAcc = &toolchainv1alpha1.UserAccount{}
```

Comment:
Check that User/Identity is deleted after the first reconcile. Also check the status (Terminating).
And check that the finalizer is deleted (and User/Identity are still deleted too) after the second reconcile.

Reply:
@alexeykazakov marking as resolved. the status will not be "terminating" here. Since there will be no error returned. Also since our discussion in slack both user/identity will be deleted in separate reconciles. Please re-open if i misunderstood though! 

however, failing to remove the finalizer is tested in another test and i've ensured that the status is checked and that it is "terminating"

Reply:
test for removing finalizer failing (state terminating): https://github.com/codeready-toolchain/member-operator/pull/105/files#diff-fc74d9d211758dbe5bd817320a8fe2a1R530

test for removing user/identity failing (state terminating): https://github.com/codeready-toolchain/member-operator/pull/105/files#diff-fc74d9d211758dbe5bd817320a8fe2a1R607

test for deleting both user and identity failing (state terminating): https://github.com/codeready-toolchain/member-operator/pull/105/files#diff-fc74d9d211758dbe5bd817320a8fe2a1R658